### ISSUE TITLE
FC: Implement graceful shutdown from external

### ISSUE THREAD

**Issue by Pennyzct** [kata-containers/kata-containers#470]
# Description of problem
For now, in kata, when trying to shutdown firecracker from external, we will try `SIGTERM` at the first place, then giving hypervisor 15s to terminate. And if failed, we will brutally `SIGKILL` it.
However, actually, firecracker couldn't respond to `SIGTERM` signal, which leading to container always waiting long 15s+ on exiting.
```
level=warning msg="VM still running after waiting 15s" ID=test source=virtcontainers subsystem=firecracker
```

# Proposal
Firecracker provides `SendCtrlAltDel` API action to send CTRL+ALT+DEL to the VM.
This can be used to trigger a graceful shutdown of the microVM, if the guest has support for i8042 and AT Keyboard.
_**But This action is only available for x86_64.**_

# To-do
I've created a PR there to enable a new API `PressGPIOPowerOff` to also have a graceful way to power off the VM from external on aarch64, which needs gpio pl061 emulation, etc. It's still under review~~~

**Comment by sboeuf** [comment]
Yes we might implement the same kind of thing for Cloud-Hypervisor, but we would rely on the ACPI power button (not implemented yet).

**Comment by ariel-adam** [comment]
We have a PR for this issue #471
